netutils/iperf/iperf_main.c

```diff
@@ -0,0 +1,230 @@
+ * under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <nuttx/config.h>
+#include <sys/time.h>
+#include <arpa/inet.h>
+#include <net/if.h>
+#include "netutils/netlib.h"
+#include "argtable3.h"
+#include "iperf.h"
+
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+
+#ifdef CONFIG_EXAMPLES_IPERFTEST_DEVNAME
+#  define DEVNAME CONFIG_EXAMPLES_IPERFTEST_DEVNAME
+#else
+#  define DEVNAME "wlan0"
+#endif
+
+#define IPERF_DEFAULT_PORT     5001
+#define IPERF_DEFAULT_INTERVAL 3
+#define IPERF_DEFAULT_TIME     30
+
+/****************************************************************************
+ * Private Types
+ ****************************************************************************/
+
+struct wifi_iperf_t
+{
+  struct arg_str *ip;
+  struct arg_lit *server;
+  struct arg_lit *udp;
+  struct arg_int *port;
+  struct arg_int *interval;
+  struct arg_int *time;
+  struct arg_lit *abort;
+  struct arg_end *end;
+};
+
+/****************************************************************************
+ * Private Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: iperf_showusage
+ *
+ * Description:
+ *   Show usage of the demo program and exit
+ *
+ ****************************************************************************/
+
+static void iperf_showusage(FAR const char *progname, int exitcode)
+{
+  printf("USAGE: %s [-sua] [-c <ip>] [-p <port>]\
+         [-i <interval>] [-t <time>]\n", progname);
+  printf("iperf command:\n");
+  printf("  -c, --client=<ip>  run in client mode,\
+         connecting to <host>\n");
+  printf("  -s, --server  run in server mode\n");
+  printf("  -u, --udp  use UDP rather than TCP\n");
+  printf("  -p, --port=<port>  server port to listen on/connect to\n");
+  printf("  -i, --interval=<interval>\
+         seconds between periodic bandwidth reports\n");
+  printf("  -t, --time=<time>\
+         time in seconds to transmit for (default 10 secs)\n");
+  printf("  -a, --abort  abort running iperf\n");
+  printf("\n");
+  exit(exitcode);
+}
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+int main(int argc, FAR char *argv[])
+{
+  struct wifi_iperf_t iperf_args;
+  struct iperf_cfg_t cfg;
+  struct in_addr addr;
+  int nerrors;
+
+  bzero(&addr, sizeof(struct in_addr));
+  bzero(&cfg, sizeof(cfg));
+
+  iperf_args.ip = arg_str0("c", "client", "<ip>",
```

Comment:
why not parse the command line directly by POSIX getopt function?

Reply:
I used the Argtable library instead of getopt to complete the parsing work, it simplifies command-line parsing by defining a declarative-style API that you can use to specify what your command-line syntax looks like, the parsing itself is done using GNU getopt so the parser is 100% GNU compatible.
However, using getopt directly will make the parsing logic a bit more complicated.

Reply:
Ok.